mongo/client.go

```diff
@@ -1021,7 +1023,7 @@ func (c *Client) ListDatabaseNames(ctx context.Context, filter interface{}, opts
 
 // WithSession creates a new SessionContext from the ctx and sess parameters and uses it to call the fn callback. The
 // SessionContext must be used as the Context parameter for any operations in the fn callback that should be executed
-// under the session.
+// under the session. WithSession is not thread safe or fork safe.
```

Comment:
`WithSession` is actually safe to call from multiple goroutines, but the the `SessionContext` passed to the callback func is not. Also, prefer Go-specific terminology for describing concurrency to prevent confusion.

For example (using concurrency terminology borrowed from [here](https://pkg.go.dev/math/rand#NewSource)):
> WithSession is safe to call from multiple goroutines concurrently. However, the SessionContext passed to the WithSession callback function is not safe for concurrent use by multiple goroutines.